outline/tun2socks/tunnel.go

```diff
@@ -40,7 +44,7 @@ type Tunnel interface {
 }
 
 // Deprecated: use Tunnel directly.
-type OutlineTunnel = Tunnel
+type OutlineTunnel Tunnel
 
 type outlinetunnel struct {
 	tunnel.Tunnel
```

Comment:
Let's remove this. I don't really think it's needed, and it's extremely confusing. Let's see what breaks and fix it. I believe the comments here will address most issues.
```suggestion
```